src/bundle/Controller/ContentViewController.php

```diff
@@ -59,6 +60,31 @@ public function detailsTabAction(ContentView $view)
         return $view;
     }
 
+    public function locationsTabAction(ContentView $view)
+    {
+        $versionInfo = $view->getContent()->getVersionInfo();
+        $contentInfo = $versionInfo->getContentInfo();
+
+        if ($contentInfo->published) {
+            $locationRepository = $this->getRepository()->getLocationService();
+            $locations = $locationRepository->loadLocations($contentInfo);
```

Comment:
For readability, I'd refactor the loading of locations to a `loadLocations` private method (or an external service if you want to, it would make it testable).